src/assertDeepEqual.lua

```diff
@@ -6,7 +6,7 @@
 	This should only be used in tests.
 ]]
 
-local function deepEqual(a, b)
+local function deepEqual(a, b): (boolean, string?)
```

Comment:
This should also work and more precisely express the scenarios:
```suggestion
local function deepEqual(a, b): (true, nil) | (false, string)
```

Reply:
Had no idea we could use boolean literals. Awesome!